src/services/notificationService.ts

```diff
@@ -106,40 +81,30 @@ export default class NotificationService {
   ): Promise<
     IResponseErrorInternal | IResponseSuccessJson<SuccessResponse>
   > => {
-    const azureInstallation: IInstallation = {
+    const azureInstallation: NotificationHubCreateOrUpdateMessage = {
       // When a single active session per user is allowed, the installation that must be created or updated
       // will have an unique installationId referred to that user.
-      // Otherwise the installationId provided by the client will be used.
+      // The installationId provided by the client is ignored.
       installationId: toFiscalCodeHash(fiscalCode),
+      kind:
+        CreateOrUpdateKind[NotificationHubMessageKindEnum.CreateOrUpdateKind],
       platform: installation.platform,
       pushChannel: installation.pushChannel,
-      tags: [toFiscalCodeHash(fiscalCode)],
-      templates: {
-        template:
-          installation.platform === PlatformEnum.apns
-            ? APNSTemplate
-            : GCMTemplate
-      }
+      tags: [toFiscalCodeHash(fiscalCode)]
     };
     return new Promise<
       IResponseErrorInternal | IResponseSuccessJson<SuccessResponse>
     >(resolve => {
-      this.notificationHubService.createOrUpdateInstallation(
-        {
-          ...azureInstallation,
-          tags: [...azureInstallation.tags]
-        },
-        (error, _) =>
+      this.notificationHubQueueClient
```

Comment:
since we don't have a callback anymore, why we still need to call resolve ?

Reply:
Fixed in c925ce9